File: top-interview-150/c++/205_isomorphic-strings.cpp
class Solution {
public:
    vector<string> strListToNumList(vector<string> s) {
        vector<string> tmp{};
        for (int i=0; i<s.size(); i++) {
            if (count(tmp.begin(), tmp.end(), s[i]) == 0) {
                tmp.push_back(s[i]);
            }
        }
        for (int i=0; i<tmp.size(); i++) {
            int x=0;
            while (x<s.size()) {
                if (s[x]==tmp[i]) {
                    s[x]=to_string(i);
                }
                x++;
            }
        }
        return s;
    }

    vector<string> stringToStringVector(string s) {
        vector<string> ret{};
        for (int i=0; i<s.size(); i++) {
            string t(1, s[i]);
            ret.push_back(t);
        }
        return ret;
    }

    bool isIsomorphic(string s, string t) {
        vector<string> ls=stringToStringVector(s);
        vector<string> lt=stringToStringVector(t);

        vector<string> lsn=strListToNumList(ls);
        vector<string> ltn=strListToNumList(lt);

        for (int i=0; i<ls.size(); i++) {
            if (lsn[i] != ltn[i]) return false;
        }
        return true;
    }
};

File: top-interview-150/python/205_isomorphic-strings.py
class Solution:
    def strListToNumList(self, s: list) -> list:
        tmp = []
        for i in range(0, len(s)):
            if s[i] not in tmp:
                tmp.append(s[i])
        #print("tmp:", tmp)
        for i in range(0, len(tmp)):
            x = 0
            while x < len(s):
                if s[x] == tmp[i]:
                    s[x] = str(i)
                x += 1
        #print("s:", s)
        return s

    def isIsomorphic(self, s: str, t: str) -> bool:
        lp = [*s]
        #print(lp)
        ls = [*t]
        #print(ls)
        if len(lp) != len(ls):
            return False
        ret1 = self.strListToNumList(lp)
        ret2 = self.strListToNumList(ls)
        for i in range(0, len(ret1)):
            if ret1[i] != ret2[i]:
                return False
        return True
